core/src/main/java/org/jdbi/v3/core/Jdbi.java

```diff
@@ -301,7 +301,7 @@ public Handle open() {
             final long start = System.nanoTime();
             @SuppressWarnings("PMD.CloseResource")
             Connection conn = Objects.requireNonNull(connectionFactory.openConnection(),
-                    () -> "Connection factory " + connectionFactory + " returned a null connection");
```

Comment:
I recall this was done as a micro-optimization to prevent all the string concatenation work when the error isn't thrown